spec/reports/_base-report.spec.js

```diff
@@ -0,0 +1,74 @@
+var _ = require('lodash');
+var _baseReport = require('../../reports/_baseReport.js');
+
+describe("tests/_baseReport", function () {
+  describe('configuration', function() {
+    it('should create a config property when no config is passed in', function() {
+      var FakeReporter = _baseReport();
+      var fakeReporter = new FakeReporter();
+
+      assert(_.isEqual(fakeReporter.config, {}));
+    });
+    it('should create a config property when only runtimeConfig is passed in', function() {
+      var runtimeTestConfig = { runtime: true };
+
+      var FakeReporter = _baseReport();
+      var fakeReporter = new FakeReporter(runtimeTestConfig);
+
+      assert(_.isEqual(fakeReporter.config, runtimeTestConfig));
+    });
+    it('should create a config property when only baseConfig is passed in', function() {
+      var baseTestConfig = { base: true };
+
+      var FakeReporter = _baseReport(baseTestConfig);
+      var fakeReporter = new FakeReporter();
+
+      assert(_.isEqual(fakeReporter.config, baseTestConfig));
+    });
+    it('should not modify the config objects', function() {
+      var baseTestConfig = {
+        base: true,
+        deep: {
+          common: "base",
+          bass: true
+        }
+      };
+      var runtimeTestConfig = {
+        runtime: true,
+        deep: {
+          common: "runtime",
+          runtime: true
+        }
+      };
+      var baseTestConfigCopy = JSON.parse(JSON.stringify(baseTestConfig));
+      var runtimeTestConfigCopy = JSON.parse(JSON.stringify(runtimeTestConfig));
+
+      var FakeReporter = _baseReport(baseTestConfig);
+      var fakeReporter = new FakeReporter(runtimeTestConfig);
+
+      assert(_.isEqual(baseTestConfig, baseTestConfigCopy));
+      assert(_.isEqual(runtimeTestConfig, runtimeTestConfigCopy));
+    });
+    it.skip('should contain unique (deep) values from both dictionaries', function() {
```

Comment:
_.extend is does not extend deep objects so right now this test fails since the _baseTest config object is creates with a regular extend. Not sure if this is true of lodash but in underscore when you extend a deep object the new one has references to the old ones. This is even true when you use the empty object as a first param which can cause some really weird stuff to happen.

Lodash's [merge](https://lodash.com/docs#merge) would solve this.